reference-adapters/ifms-adapter/src/main/java/org/egov/service/PAGService.java

```diff
@@ -0,0 +1,131 @@
+package org.egov.service;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import net.minidev.json.JSONArray;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.repository.PIRepository;
+import org.egov.tracer.model.CustomException;
+import org.egov.utils.BillUtils;
+import org.egov.utils.HelperUtil;
+import org.egov.utils.PIUtils;
+import org.egov.web.models.bill.Payment;
+import org.egov.web.models.bill.PaymentRequest;
+import org.egov.web.models.enums.*;
+import org.egov.web.models.jit.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.egov.config.Constants.JIT_BILL_DATE_FORMAT;
+
+@Service
+@Slf4j
+public class PAGService {
+
+    @Autowired
+    private PaymentInstructionService paymentInstructionService;
+    @Autowired
+    private IfmsService ifmsService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private PIRepository piRepository;
+
+    @Autowired
+    private HelperUtil helperUtil;
+
+    @Autowired
+    private PIUtils piUtils;
+
+    @Autowired
+    private BillUtils billUtils;
+
+    public void updatePAG( RequestInfo requestInfo ){
+        List<PaymentInstruction> paymentInstructions = getApprovedPaymentInstructions();
```

Comment:
There are no logging statements in this service. Please add INFO and fine grained logs to enable debug.